packages/back-end/src/models/DataSourceModel.ts

```diff
@@ -115,6 +115,10 @@ export async function createDataSource(
     });
   }
 
+  settings?.queries?.exposure?.map(
+    (exposureObj) => (exposureObj.hasNameCol = false)
+  );
```

Comment:
This will overwrite the property for all new data sources, which doesn't seem correct.  What if we added that toggle to the initial create form?  This code would stop that from working properly.  I think we can just remove this code completely.

Reply:
I think I wanted each data source by default to have the hasNameCol property set to false, which is why I have this code here. Yeah, this was for the toggle to work later, so that the hasNameCol is defined and has a set value for each exposure query object.

Reply:
Resolved by new changes